### ISSUE TITLE
`az --version` performs an update check

### ISSUE THREAD

**Issue by jessehouwing** [Azure/azure-cli#27437]
### Describe the bug

When running `az --version` an update check is performed against az-cli and the extensions.

Azure Pipelines' AzureCLI task runs `az --version` to capture the version details in the logs.

It would save quite a bit of CPU if az-cli would not perform an update check in a CI/CD scenario.

There already is an core.auto-upgrade=false option, would it not make sense to piggy back on that or to add another config setting to turn off update checking.

### Related command

`az --version`

### Errors

N/a

### Issue script & Debug output

N/a

### Expected behavior

No update check is performed in CI.

### Environment Summary

2.52.0

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by jessehouwing** [comment]
Yeah, have a way to disable network access for these commands. Either with an extra parameter to `--suppress-update-check` or a setting in the config file or an environment variable.

Ideally `az` would detect it's in CI/CD and turn off a load of stuff that only adds logspam and slowdowns by default.

It looks like `az init` is making a start of collecting the desired settings ;).

**Comment by jiasli** [comment]
`az init` is actually a secret project. May I know where you heard of it? 🤣

**Comment by jessehouwing** [comment]
> `az init` is actually a secret project. May I know where you heard of it? 🤣

I was looking for a way to set the default config settings, and I tried it. No and behold it downloaded the extension and ran.

I supplied a PR for it to add a few more settings.

**Comment by jiasli** [comment]
> Ideally `az` would detect it's in CI/CD and turn off a load of stuff that only adds logspam and slowdowns by default.

This is another topic we are still internally investigating. Not only Azure DevOps, we are aiming to find a unified way to detect the running environment - Azure DevOps, Azure Cloud Shell, GitHub Actions, etc.

**Comment by jiasli** [comment]
`az init` is only an experimental feature we made in https://github.com/Azure/azure-cli-extensions/pull/3969. It is never expected to be used in production. Also, `az init` allows the user to _select_ the running environment. It can't really _detect_ the running environment.

**Comment by jessehouwing** [comment]
> `az init` is only an experimental feature we made in [Azure/azure-cli-extensions#3969](https://github.com/Azure/azure-cli-extensions/pull/3969). It is never expected to be used in production. Also, `az init` allows the user to _select_ the running environment. It can't really _detect_ the running environment.

Yeah, I figured as much. The experimental nature at least. But it does seem to reccomend certain settings in an automated environment.

As you mentioned above `auto-upgrade.enable=yes` in the docs turns out to probably be `auto-upgrade.enable=true` in the code, at least there are quite a few issues that dispute the correct syntax. I went looking for what settings there are and what their values should/could be and there is VERY little documentation on those topics.

I ended up doing a search through the codebase instead. But it lead to some useful insights! 

I've been digging into the performance of `az` and `az devops` in Azure Pipelines and GtHub Hosted Runners. 

 - https://github.com/actions/runner-images/pull/8294
 - https://github.com/microsoft/azure-pipelines-tasks/issues/19006
 - https://github.com/microsoft/azure-pipelines-tasks/issues/19007
 - https://github.com/Azure/azure-cli/issues/27444
 - https://github.com/Azure/azure-cli/issues/27404
 - https://github.com/Azure/azure-devops-cli-extension/issues/1366
 - https://github.com/Azure/azure-cli-extensions/pull/3969
 - https://github.com/Azure/azure-devops-cli-extension/issues/1367
 
If these each shave off a couple of seconds, over the number of invocations on the hosted runners worldwide it would save a couple of trees and a bunch of water.

Ideally `az` would detect it's in a CI/CD environment and turn off:

 - Telemetry
 - Disk logging
 - Error recommendations
 - Colors
 - Auto-updates
 - Version checks
 - Survey banners
 - auto-installation of missing extensions
 - Randomly downloading new pip packages (like keyring) or helm versions or other unexpectedly missing dependencies. I'd rather have it fail in CI/CD than for it to download versions of files I haven't been able to vet. 
 -
